lib/Router.js

```diff
@@ -234,7 +235,10 @@ Router.prototype.handle = function(request, connection) {
 
         if (props.length) {
             route.handlers[method].filter(function(handler) {
-                return _.intersection(request.getProps(), handler.props).length;
+                var topLevelProps = props.map(function(prop) {
+                    return prop.split('.')[0];
+                });
+                return _.intersection(topLevelProps, handler.props).length;
```

Comment:
Missing top-level missing properties are handled:

    $ curl http://localhost:5150/my-api/users/1?props=age
    {
      "$id": "/users/1",
      "$expires": 3600,
      "age": 27
    }
    $ curl http://localhost:5150/my-api/users/1?props=unknown
    {
      "code": 404,
      "error": "NOT FOUND",
      "message": "No handlers available for resource /users/1",
      "properties": [],
      "$httpStatus": 404
    }